resharper/resharper-unity/src/Rider/Protocol/BackendUnityHost.cs

```diff
@@ -69,7 +69,13 @@ public class BackendUnityHost
                 // (not actually passthrough)
                 var frontendBackendModel = frontendBackendHost.Model.NotNull("frontendBackendHost.Model != null");
                 BackendUnityModel.FlowIntoRdSafe(lifetime,
-                    backendUnityModel => backendUnityModel != null,
+                    backendUnityModel =>
+                    {
+                        var res = backendUnityModel != null;
+                        // when connection to Unity disappears, state of PlayControls in the Frontend-Backend protocol may not correspond to state in Unity
+                        if (!res) frontendBackendHost.Model?.PlayControlsInitialized.SetValue(false);
+                        return res;
+                    },
```

Comment:
This should be moved to the passthrough host. The backend/unity host should not have a reference to the frontend/backend as it makes it much easier to introduce circular dependencies. The passthrough host is where all other frontend <-> unity communication happens.

The backend/unity host only has a reference to the frontend host to handle the editor state polling, which should really be handled with a property and should really be part of the passthrough host (I left it here because it felt more like infrastructure than passing data around, but it's still passthrough stuff)